File: app/src/main/java/io/github/alexeychurchill/stickynotes/fragment/IncomingFriendRequestListFragment.java
package io.github.alexeychurchill.stickynotes.fragment;

import android.view.View;

import java.util.List;

import io.github.alexeychurchill.stickynotes.R;
import io.github.alexeychurchill.stickynotes.model.FriendRequest;
import io.github.alexeychurchill.stickynotes.model.ServiceResponse;
import retrofit2.Call;

/**
 * Incoming friend list request
 */

public class IncomingFriendRequestListFragment extends FriendRequestListFragment {
    private String mActionAcceptTitle = "";
    private String mActionRejectTitle = "";

    @Override
    public String getActionOneTitle() { // Accept
        return mActionAcceptTitle;
    }

    @Override
    public String getActionTwoTitle() { // Reject
        return mActionRejectTitle;
    }

    @Override
    public boolean userListNeedActionOne() {
        return true;
    }

    @Override
    public boolean userListNeedActionTwo() {
        return true;
    }

    @Override
    protected Call<ServiceResponse<Object>> getActionOneCall(FriendRequest friendRequest) {
        if (getApi() == null || getAccessToken() == null) {
            return null;
        }
        return getApi().friendAcceptRequest(getAccessToken(), friendRequest.getId());
    }

    @Override
    protected Call<ServiceResponse<Object>> getActionTwoCall(FriendRequest friendRequest) {
        if (getApi() == null || getAccessToken() == null) {
            return null;
        }
        return getApi().friendDeleteRequest(getAccessToken(), friendRequest.getId());
    }

    @Override
    protected void onInit(View view) {
        super.onInit(view);
        // Actions titles
        mActionAcceptTitle = getString(R.string.text_button_accept);
        mActionRejectTitle = getString(R.string.text_button_reject);
    }

    @Override
    protected Call<ServiceResponse<List<FriendRequest>>> getLoadDataPageCall(int page) {
        if (getApi() == null || getAccessToken() == null) {
            return null;
        }
        return getApi().friendGetRequests(getAccessToken(), page);
    }
}

File: app/src/main/java/io/github/alexeychurchill/stickynotes/contacts/IncomingFriendRequestListFragment.kt
package io.github.alexeychurchill.stickynotes.contacts

import android.view.View
import io.github.alexeychurchill.stickynotes.R
import io.github.alexeychurchill.stickynotes.model.FriendRequest
import io.github.alexeychurchill.stickynotes.model.ServiceResponse
import retrofit2.Call

/**
 * Incoming friend list request
 */
class IncomingFriendRequestListFragment : FriendRequestListFragment() {
    private var mActionAcceptTitle = ""
    private var mActionRejectTitle = ""
    override fun getActionOneTitle(): String { // Accept
        return mActionAcceptTitle
    }

    override fun getActionTwoTitle(): String { // Reject
        return mActionRejectTitle
    }

    override fun userListNeedActionOne(): Boolean {
        return true
    }

    override fun userListNeedActionTwo(): Boolean {
        return true
    }

    override fun getActionOneCall(friendRequest: FriendRequest?): Call<ServiceResponse<Any>>? {
        return if (api == null || accessToken == null) {
            null
        } else api!!.friendAcceptRequest(accessToken, friendRequest!!.id)
    }

    override fun getActionTwoCall(friendRequest: FriendRequest?): Call<ServiceResponse<Any>>? {
        return if (api == null || accessToken == null) {
            null
        } else api!!.friendDeleteRequest(accessToken, friendRequest!!.id)
    }

    override fun onInit(view: View?) {
        super.onInit(view)
        // Actions titles
        mActionAcceptTitle = getString(R.string.text_button_accept)
        mActionRejectTitle = getString(R.string.text_button_reject)
    }

    override fun getLoadDataPageCall(page: Int): Call<ServiceResponse<List<FriendRequest>>>? {
        return if (api == null || accessToken == null) {
            null
        } else api!!.friendGetRequests(accessToken, page)
    }
}